src/test/java/com/google/devtools/build/lib/remote/DiskCacheIntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+package com.google.devtools.build.lib.remote;
+
+import static com.google.devtools.build.lib.testutil.TestUtils.tmpDirFile;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.buildtool.util.BuildIntegrationTestCase;
+import com.google.devtools.build.lib.runtime.BlazeModule;
+import com.google.devtools.build.lib.runtime.BlazeRuntime;
+import com.google.devtools.build.lib.runtime.BlockWaitingModule;
+import com.google.devtools.build.lib.runtime.BuildSummaryStatsModule;
+import com.google.devtools.build.lib.standalone.StandaloneModule;
+import com.google.devtools.build.lib.vfs.PathFragment;
+import java.io.IOException;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DiskCacheIntegrationTest extends BuildIntegrationTestCase {
+
+  private static PathFragment getDiskCacheDir() {
+    PathFragment testTmpDir = PathFragment.create(tmpDirFile().getAbsolutePath());
+    return testTmpDir.getRelative("disk_cache");
+  }
+
+  @Override
+  protected void setupOptions() throws Exception {
+    super.setupOptions();
+
+    addOptions("--disk_cache=" + getDiskCacheDir().toString());
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    getWorkspace().getFileSystem().getPath(getDiskCacheDir()).deleteTree();
+  }
+
+  @Override
+  protected ImmutableList<BlazeModule> getSpawnModules() {
+    return ImmutableList.<BlazeModule>builder()
+        .addAll(super.getSpawnModules())
+        .add(new StandaloneModule())
+        .build();
+  }
+
+  @Override
+  protected BlazeRuntime.Builder getRuntimeBuilder() throws Exception {
+    return super.getRuntimeBuilder()
+        .addBlazeModule(new RemoteModule())
+        .addBlazeModule(new BuildSummaryStatsModule())
+        .addBlazeModule(new BlockWaitingModule());
+  }
+
+  @Test
+  public void hitDiskCache() throws Exception {
+    write("BUILD",
+        "genrule(",
+        "  name = 'foo',",
+        "  srcs = ['foo.in'],",
+        "  outs = ['foo.out'],",
+        "  cmd = 'cat $(SRCS) > $@',",
+        ")",
+        "genrule(",
+        "  name = 'foobar',",
+        "  srcs = [':foo.out', 'bar.in'],",
+        "  outs = ['foobar.out'],",
+        "  cmd = 'cat $(SRCS) > $@',",
+        ")"
+    );
+    write("foo.in", "foo");
+    write("bar.in", "bar");
+    buildTarget("//:foobar");
+    cleanAndRestartServer();
+
+    buildTarget("//:foobar");
+
+    events.assertContainsInfo("2 disk cache hit");
+  }
+
+  private void blobsReferencedInACAreMissingCASIgnoresAc(Runnable setupOptions) throws Exception {
```

Comment:
Taking a runnable here seems unnecessarily complex - a `String...` would do? (Perhaps you plan to use it in a more complex way later, in which case it's fine.)

Reply:
Changed to `String...`